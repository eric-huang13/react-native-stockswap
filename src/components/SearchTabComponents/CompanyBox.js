import React, {Component} from 'react';
import {
  Text,
  View,
  StyleSheet,
  SafeAreaView,
  Dimensions,
  PixelRatio,
  Platform,
} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import {connect} from 'react-redux';
import {moderateScale, scale} from '../../util/responsiveFont';

export class CompanyBox extends Component {


  
  render() {
    const {item, category} = this.props;
    const {width, height} = Dimensions.get('window');
    const symbol = item.ticker   
    const stockObject =this.props.tickersAll[symbol]
    const styledText =
      category === 'gainers' ? (
        <LinearGradient
          start={{x: 0.1, y: 0.1}}
          end={{x: 1, y: 1}}
          colors={['#2C3957', '#2C572E']}
          style={style.linearGradient}>
          <SafeAreaView style={style.listContainer}>
            <View style={style.topDetails}>
              <Text style={{...style.symbol, color: '#1AB968'}}>
                {item.ticker}
              </Text>
              { stockObject.name ?
           <Text style={style.title}>
                {stockObject.name.length < 15
                  ? `${stockObject.name}`
                  : `${stockObject.name.substring(0, 14)}...`}
              </Text>
              :
              null
  }
            </View>
            <View style={style.bottomDetails}>
              <Text style={style.price}>${item.quote.volumeWeightedAveragePrice}</Text>
              <Text style={{...style.percentage, color: '#1AB968'}}>
               +{item.change}%
              </Text>
            </View>
          </SafeAreaView>
        </LinearGradient>
      ) : category === 'losers' ? (
        <LinearGradient
          start={{x: 0.1, y: 0.1}}
          end={{x: 1, y: 1}}
          colors={['#2C3957', '#572C3D']}
          style={style.linearGradient}>
          <SafeAreaView style={style.listContainer}>
            <View style={style.topDetails}>
              <Text style={{...style.symbol, color: '#D13C3D'}}>
                {item.symbol}
              </Text>
              <Text style={style.title}>
                {item.title.length < 15
                  ? `${item.title}`
                  : `${item.title.substring(0, 14)}...`}
              </Text>
            </View>
            <View style={style.bottomDetails}>
              <Text style={style.price}>${item.price}</Text>
              <Text style={{...style.percentage, color: '#D13C3D'}}>
                {item.percentage}
              </Text>
            </View>
          </SafeAreaView>
        </LinearGradient>
      ) :  (
        <LinearGradient
          start={{x: 0.1, y: 0.1}}
          end={{x: 1, y: 1}}
          colors={['#2C3957', '#5B449B']}
          style={style.linearGradient}>
          <SafeAreaView style={style.listContainer}>
            <View style={style.topDetails}>
              <Text style={{...style.symbol, color: '#B8A0FF'}}>
                {item.symbol}
              </Text>
              <Text style={style.title}>
                {item.title.length < 15
                  ? `${item.title}`
                  : `${item.title.substring(0, 14)}...`}
              </Text>
            </View>
            <View style={style.bottomDetails}>
              <Text style={style.price}>${item.price}</Text>
              <Text style={{...style.percentage, color: '#1AB968'}}>
                {item.percentage}
              </Text>
            </View>
          </SafeAreaView>
        </LinearGradient>
      );

    return <SafeAreaView>{styledText}</SafeAreaView>;
  }
}

const mapStateToProps = (state) => {
  return {
    tickersAll: state.company.tickersAll
  };
};
const mapDispatchToProps = (dispatch) => {
  return {
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(CompanyBox);
// export default CompanyBox;

const style = StyleSheet.create({
  linearGradient: {
    alignSelf: 'center',
    marginTop: moderateScale(3),
    borderRadius: moderateScale(12),
    height: Dimensions.get('window').height / 3 - 108,
    width: Dimensions.get('window').width / 3 - 10,
    //      height: 120,
    //      width: 127,
    height: scale(120),
    width: scale(127),

    flexDirection: 'column',
    padding: moderateScale(3),
    paddingTop: 0,
    justifyContent: 'space-around',
    marginLeft: moderateScale(4),
    marginRight: moderateScale(4),
  },
  listContainer: {
    borderRadius: moderateScale(15),
    flex: 1,
    flexDirection: 'column',
    paddingHorizontal: moderateScale(4),
    paddingTop: 0,
    justifyContent: 'space-evenly',
  },
  topDetails: {
    marginTop: moderateScale(-4),
  },
  bottomDetails: {},
  title: {
    fontSize: moderateScale(13),
    fontFamily: 'Montserrat-Regular',
    color: '#b3b3b3',
  },
  detailsContainer: {
    position: 'absolute',
    marginTop: '60%',
    marginLeft: moderateScale(2),
  },
  symbol: {
    fontSize: moderateScale(19),
    color: 'rgb(8, 11, 9)',
    fontFamily: 'Montserrat-Bold',
  },
  percentage: {
    fontSize: moderateScale(13),
    color: 'grey',
    fontFamily: 'Montserrat-SemiBold',
  },
  price: {
    fontSize: moderateScale(19),
    color: '#FFFFFF',
    fontFamily: 'Montserrat-Regular',
  },
});
