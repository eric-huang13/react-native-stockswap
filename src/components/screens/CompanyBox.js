import React, { Component } from 'react'
import {Text, View,} from 'react-native';
import {companyBoxStyles} from '../../styles/companyBoxStyles';



export class CompanyBox extends Component {
    render() {
        const {item} = this.props;

        return (
            <View>
                <View  style={ item.category === "gainers" ? {...companyBoxStyles.listContainer, backgroundColor:'rgb(8, 177, 40)'} : item.category === "losers" ? {...companyBoxStyles.listContainer, backgroundColor:'rgb(196, 38, 0)'} : {...companyBoxStyles.listContainer, backgroundColor:'rgb(58, 117, 167)'} } key={item.id}>
               <Text style={companyBoxStyles.title}>{item.title}</Text>
                  <View style={companyBoxStyles.detailsContainer}>
                    <Text style={companyBoxStyles.symbol}>{item.symbol}</Text>
                    <Text style={companyBoxStyles.percentage}>
                      {item.percentage}
                    </Text>
                  </View>
              </View>
            </View>
        )
    }
}

export default CompanyBox

