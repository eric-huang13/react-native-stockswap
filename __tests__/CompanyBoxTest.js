import React from 'react';
import CompanyBox from '../src/components/SearchTabComponents/CompanyBox';
import ShallowRenderer from 'react-test-renderer/shallow'; 


jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock"),
);
jest.mock('react-redux', () => ({
    connect: () => {
      return (component) => {
        return component
      };
    }
  }));

test('renders correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<CompanyBox item={{title:'test' }}/>);
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});