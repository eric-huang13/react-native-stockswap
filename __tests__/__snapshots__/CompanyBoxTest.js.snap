// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ForwardRef(SafeAreaView)>
  <LinearGradient
    colors={
      Array [
        "#2C3957",
        "#5B449B",
      ]
    }
    end={
      Object {
        "x": 1,
        "y": 1,
      }
    }
    start={
      Object {
        "x": 0.1,
        "y": 0.1,
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "borderRadius": 16.94890510948905,
        "flexDirection": "column",
        "height": 218.97810218978103,
        "justifyContent": "space-around",
        "marginLeft": 5.649635036496351,
        "marginRight": 5.649635036496351,
        "marginTop": 4.237226277372263,
        "padding": 4.237226277372263,
        "paddingTop": 0,
        "width": 231.75182481751824,
      }
    }
  >
    <ForwardRef(SafeAreaView)
      style={
        Object {
          "borderRadius": 21.186131386861312,
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-evenly",
          "paddingHorizontal": 5.649635036496351,
          "paddingTop": 0,
        }
      }
    >
      <View
        style={
          Object {
            "marginTop": -5.649635036496351,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#B8A0FF",
              "fontFamily": "Montserrat-Bold",
              "fontSize": 26.835766423357665,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#b3b3b3",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 18.361313868613138,
            }
          }
        >
          test
        </Text>
      </View>
      <View
        style={Object {}}
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 26.835766423357665,
            }
          }
        >
          $
        </Text>
        <Text
          style={
            Object {
              "color": "#1AB968",
              "fontFamily": "Montserrat-SemiBold",
              "fontSize": 18.361313868613138,
            }
          }
        />
      </View>
    </ForwardRef(SafeAreaView)>
  </LinearGradient>
</ForwardRef(SafeAreaView)>
`;
